test/integration/network_reserved_ips_test.go

```diff
@@ -2,150 +2,199 @@ package integration
-		if fetchIpErr != nil {
-			t.Logf("Error getting ipaddress, expected struct, got %v and error %v", reservedIP, fetchIpErr)
-		}
+	if resIP == nil {
+		t.Fatalf("Reserved IP is nil")
+	}
 
-		// Verify the list of IPs has increased
-		verifyList, verifyListIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-		if verifyListIpErr != nil {
-			t.Logf("Error listing ipaddresses, expected struct, got error %v", verifyListIpErr)
-		} else {
-			if len(verifyList)-len(ipList) == 1 {
-				t.Log("Increase in IP list confirmed", verifyList)
-			} else {
-				t.Errorf("Increase in IP list not confirmed")
-			}
-		}
+	t.Logf("Successfully reserved IP: %+v", resIP)
 
-		// Delete the reserved IP
-		if reserveIP != nil {
-			deleteErr := client.DeleteReservedIPAddress(context.Background(), reserveIP.Address)
-			if deleteErr != nil {
-				t.Logf("Error deleting reserved IP address: %v", deleteErr)
-			} else {
-				t.Logf("Successfully deleted reserved IP address: %s", reserveIP.Address)
-			}
-		}
+	// Fetch the reserved IP
+	fetchedIP, fetchErr := client.GetReservedIPAddress(context.Background(), resIP.Address)
+	if fetchErr != nil {
+		t.Errorf("Error getting reserved IP address: %v", fetchErr)
+	}
 
-		// Verify the IP has been deleted
-		if reserveIP != nil {
-			verifyDeletedIP, fetchDeletedIpErr := client.GetReservedIPAddress(context.Background(), reserveIP.Address)
-			if fetchDeletedIpErr != nil {
-				t.Logf("Error getting ipaddress, expected struct, got %v and error %v", verifyDeletedIP, fetchDeletedIpErr)
-			}
+	if fetchedIP == nil {
+		t.Errorf("Retrieved reserved IP is nil")
+	}
 
-			verifyDeletedFromList, verifyDeletedFromListIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-			if verifyDeletedFromListIpErr != nil {
-				t.Logf("Error listing ipaddresses, expected struct, got error %v", verifyDeletedFromListIpErr)
-			} else {
-				if len(verifyDeletedFromList) < len(verifyList) {
-					t.Log("IP address deletion confirmed", verifyDeletedFromList)
-				} else {
-					t.Errorf("Verification - Failed")
-				}
-			}
-		}
+	// Verify the list of IPs has increased
+	verifyList, verifyErr := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+	if verifyErr != nil {
+		t.Fatalf("Error listing IP addresses after reservation: %v", verifyErr)
+	}
+
+	if len(verifyList) != initialCount+1 {
+		t.Errorf("Expected IP count to increase by 1, got %d, want %d", len(verifyList), initialCount+1)
+	}
+
+	// Delete the reserved IP
+	delErr := client.DeleteReservedIPAddress(context.Background(), resIP.Address)
+	if delErr != nil {
+		t.Fatalf("Error deleting reserved IP address: %v", delErr)
+	}
+
+	// Verify the IP has been deleted
+	_, fetchDelErr := client.GetReservedIPAddress(context.Background(), resIP.Address)
+	if fetchDelErr == nil {
+		t.Errorf("Expected error when fetching deleted IP, got nil")
```

Comment:
Same as prev. comment about which IP was expected.

Reply:
I have now included the IP that I am trying to fetch unsuccessfully in the error message.